accounting_application/Dockerfile

```diff
@@ -0,0 +1,17 @@
+# Use the official Node.js base image
+FROM node:14-alpine
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the NestJS application files
+COPY ./accounting_application /app
+
+# Install dependencies
+RUN npm install
+
+# Expose the port the app runs on
+EXPOSE 3001
+
+# Run the application
+CMD ["npm", "start"]
```

Comment:
Handle building the app in Docker, I don't want to install any dependencies on the EC2 except for docker.
Something like this would work.

```Dockerfile
FROM node:alpine AS development

WORKDIR /usr/src/app

COPY package.json ./

COPY yarn.lock ./

RUN yarn

COPY . . 

RUN yarn build

FROM node:alpine as production

ARG NODE_ENV=production
ENV NODE_ENV=${NODE_ENV}

WORKDIR /usr/src/app

COPY package.json ./

COPY yarn.lock ./

RUN yarn --only=prod

COPY . .

COPY --from=development /usr/src/app/dist ./dist

CMD ["yarn", "prod"]

```

Reply:
ok i will deal with it